### PR TITLE
treewide: move fixDarwinDylibNames to nativeBuildInputs

### PR DESCRIPTION
**Pull request by thefloweringash**
###### Motivation for this change

Hopefully simple cleanup. I discovered some inconsistencies when cross compiling where install_name_tool has a target prefix, and checked through the uses in the entire tree.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/security/sudolikeaboss/default.nix [hunk 1]
```diff
@@ -19,9 +19,11 @@ buildGoPackage rec {
 
   propagatedBuildInputs = with darwin.apple_sdk.frameworks; [
     Cocoa
-    fixDarwinDylibNames
   ];
 
+  propagatedNativeBuildInputs =
```
Comment (LnL7) [inline pkgs/tools/security/sudolikeaboss/default.nix:24]: This should never be propagated, that makes it impossible to build libraries rpath or executable relative references when this is a dependency.

**Review by LnL7 — CHANGES_REQUESTED** [review]
Looks good to me otherwise.

**Comment by FRidh** [comment]
needs a rebase

**Comment by thefloweringash** [comment]
Rebased!

**Review by LnL7 — APPROVED** [review]

**Pull request merged**